test/feature-benchmark/mzcompose.py

```diff
@@ -378,31 +372,31 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
 
     c.up(*dependencies)
 
-    scenarios = initial_scenarios.copy()
-
     for cycle in range(0, args.max_retries):
         print(
-            f"Cycle {cycle+1} with scenarios: {', '.join([scenario.__name__ for scenario in scenarios.keys()])}"
+            f"Cycle {cycle+1} with scenarios: {', '.join([scenario.__name__ for scenario in scenarios])}"
         )
 
         report = Report()
 
-        for scenario in list(scenarios.keys()):
+        new_scenarios = []
+        for scenario in scenarios:
             comparators = run_one_scenario(c, scenario, args)
             report.extend(comparators)
 
             # Do not retry the scenario if no regressions
-            if all([not c.is_regression() for c in comparators]):
-                del scenarios[scenario]
+            if any([c.is_regression() for c in comparators]):
+                new_scenarios.append(scenario)
```

Comment:
Why is this called `new_scenarios`? Can we rename it to `failed_scenarios` or `scenarios_with_regressions` or `scenarios_to_retry`?